Mathlib/CategoryTheory/ConnectedComponents.lean

```diff
@@ -36,6 +36,17 @@ variable {J : Type u₁} [Category.{v₁} J]
 def ConnectedComponents (J : Type u₁) [Category.{v₁} J] : Type u₁ :=
   Quotient (Zigzag.setoid J)
 
+/-- The map `ConnectedComponents J → ConnectedComponents K` induced by a functor `J ⥤ K`. -/
+def Functor.mapConnectedComponents {K : Type u₂} [Category.{v₂} K] (F : J ⥤ K)
+    (x : ConnectedComponents J) : ConnectedComponents K :=
+  x |> Quotient.lift (Quotient.mk (Zigzag.setoid _) ∘ F.obj)
+    (fun _ _ ↦ Quot.sound ∘ zigzag_obj_of_zigzag F)
+
+@[simp]
+lemma image_of_cc_is_cc_of_image {K : Type u₂} [Category.{v₂} K] (F : J ⥤ K) :
+    F.mapConnectedComponents ∘ (Quotient.mk (Zigzag.setoid J))
+      =  Quotient.mk (Zigzag.setoid K) ∘ F.obj := rfl
```

Comment:
I was thinking of this statement:
```suggestion
@[simp]
lemma Functor.mapConnectedComponents_mk {K : Type u₂} [Category.{v₂} K] (F : J ⥤ K) (j : J) :
    F.mapConnectedComponents (Quotient.mk _ j) = Quotient.mk _ (F.obj j) := rfl
```
(Note also that the names of the lemmas are important. Also, according to mathlib's coding style, infix operators like `=` should not appear after a linebreak: they must appear before.)